scripts/prepare_release.sh

```diff
@@ -86,7 +105,16 @@ read_command_line_arguments() {
 
 	shift $((OPTIND-1))
 
-	[[ $is_hotfix ]] && branch_name="hotfix" || branch_name="release"
+	if [[ $is_hotfix ]]; then
+		version_to_hotfix=${version}
+		if ! [[ $is_coldfix ]]; then
+			# shellcheck disable=SC2206
+			arrIN=(${version//./ })
+			patch_number=$((arrIN[2]+1))
+			version="${arrIN[0]}.${arrIN[1]}.${patch_number}"
```

Comment:
perhaps you can use IFS?

    IFS='.' read -ra arrIN <<< "$version"
    patch_number=$((arrIN[2] + 1))
    version="${arrIN[0]}.${arrIN[1]}.$patch_number"